client/src/views/Report/Sunburst/index.js

```diff
@@ -0,0 +1,223 @@
+    if (!active) {
+        return null;
+    }
+    if (payload.length <= 0) {
+        return null;
+    }
+
+    const { name, value } = payload[0];
+
+    return (
+        <div className={styles.customTooltip}>
+            <p>
+                {name}
+            </p>
+            <p className={styles.customTooltipValue}>
+                {value}
+            </p>
+        </div>
+    );
+};
+
+CustomTooltip.propTypes = {
+    active: PropTypes.bool, // eslint-disable-line react/require-default-props
+    // eslint-disable-next-line react/require-default-props
+    payload: PropTypes.array, // eslint-disable-line react/forbid-prop-types
+};
+
+export default function SunBurst(props) {
+    const {
+        data,
+        colorScheme,
+    } = props;
+
+    const [level, setLevel] = useState('zero');
+    const [key, setKey] = useState('');
+
+    const updateFilterParams = useCallback(
+        (levelParam, keyParam) => {
+            setLevel(levelParam);
+            setKey(keyParam);
+        },
+        [setLevel, setKey],
+    );
+
+    const {
+        centerPieData,
+        firstLevelData,
+        secondLevelData,
+    } = useMemo(
+        () => getFilteredData(level, key, data),
+        [level, key, data],
+    );
+
+    return (
+        <ResponsiveContainer
+            width="100%"
+            height={300}
+        >
+            <PieChart>
+                {
+                    centerPieData && (
+                        <Pie
+                            data={centerPieData}
+                            dataKey="size"
+                            outerRadius={30}
+                        >
+                            {
+                                firstLevelData.map(d => (
+                                    <Cell
+                                        key={d.key}
+                                        fill="#a6cee3"
+                                        onClick={() => updateFilterParams('zero', '')}
+                                        className={styles.sunburstCell}
+                                    />
+                                ))
+                            }
+                        </Pie>
+                    )
+                }
+                {
+                    firstLevelData && (
+                        <Pie
+                            data={firstLevelData}
+                            dataKey="size"
+                            innerRadius={centerPieData.length > 0 ? 32 : 0}
+                            outerRadius={centerPieData.length > 0 ? 60 : 40}
+                        >
+                            {
+                                firstLevelData.map(d => (
+                                    <Cell
+                                        key={d.key}
+                                        fill={getCellColor(d.key, colorScheme)}
+                                        onClick={() => updateFilterParams('first', d.key)}
```

Comment:
Use usecallback for this if possible

Reply:
getCellColor or updateFilterParams ??

useCallback used in updateFilterParams  and getCellColor is defined outside the component !!